02_maps/04/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+)
+
+type Student struct {
+	Id    int    `json:"id"`
+	Name  string `json:"name"`
+	Grade int    `json:"grade"`
+}
+
+type Object struct {
+	Id   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+type Result struct {
+	ObjectId  int `json:"object_id"`
+	StudentId int `json:"student_id"`
+	Result    int `json:"result"`
+}
+
+type Data struct {
+	Students []Student `json:"students"`
+	Objects  []Object  `json:"objects"`
+	Results  []Result  `json:"results"`
+}
+
+var Students = make(map[int]Student)
+var Objects = make(map[int]Object)
+
+func main() {
+	printAggregatedResultTable()
+}
+
+/*
+*
+4. Для предыдущей задачи необходимо вывести сводную таблицу по всем предметам
+*/
+
+func printAggregatedResultTable() {
+	data := readFile()
+	unanimousResults := make(map[int]map[int][]float64)
+
+	for _, result := range data.Results {
+		if unanimousResults[result.ObjectId] == nil {
+			unanimousResults[result.ObjectId] = make(map[int][]float64)
+		}
+		unanimousResults[result.ObjectId][Students[result.StudentId].Grade] = append(unanimousResults[result.ObjectId][Students[result.StudentId].Grade], float64(result.Result))
```

Comment:
очень сложно читается
лучше выделить память на переменную в стеке функции, чем каждый раз читать из мапы
и читаемость повысится

```suggestion
		objRes, ok := unanimousResults[result.ObjectId]
		if !ok {
			objRes = make(map[int][]float64)
		}
		
		grade := Students[result.StudentId].Grade
		objRes[grade] = append(objRes[grade], float64(result.Result))
		
		unanimousResults[result.ObjectId] = objRes
```